pkg/venafi/cloud/connector.go

```diff
@@ -757,6 +740,183 @@ func (c *Connector) SearchCertificate(zone string, cn string, sans *certificate.
+		return nil, fmt.Errorf("could not find keystore with ID: %s", keystoreIdString)
+	}
+
+	if len(data.CloudKeystores.Nodes) > 1 {
+		return nil, fmt.Errorf("found more than one keystore for ID: %s", keystoreIdString)
+	}
+
+	// grabbing found keystore for later
+	cloudKeystore := data.CloudKeystores.Nodes[0]
+	log.Printf("successfully fetched keystore information for KeystoreID: %s", keystoreIdString)
+
+	wsClientId := uuid.New()
+
+	wsClientIdString := wsClientId.String()
+
+	wsConn, err := websocket.Subscribe(c.apiKey, c.baseURL, wsClientIdString)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Provisioning Certificate ID %s for Keystore %s", certificateIdString, keystoreIdString)
+	_, err = cloudkeystores.ProvisionCertificate(ctx, graphqlClient, certificateIdString, keystoreIdString, wsClientIdString, provisioningOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	ar, err := websocket.ReadResponse(wsConn)
+	if err != nil {
+		return nil, err
+	}
+
+	// parsing metadata from websocket response
+	log.Printf("Getting Cloud Metadata of Certificate ID %s and Keystore ID: %s", certificateIdString, keystoreIdString)
+	respMap := ar.Data.Result
+	cloudMetadata, err := getCloudMetadataFromWebsocketResponse(respMap, cloudKeystore.Type, keystoreIdString)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("Successfully got Cloud Metadata for Certificate ID %s and Keystore ID: %s", certificateIdString, keystoreIdString)
+
+	log.Printf("Successfully finished Provisioning Flow for Certificate ID %s and Keystore ID %s", certificateIdString, keystoreIdString)
+	return cloudMetadata, nil
+}
+
+func (c *Connector) getCertIDFromPickupID(pickupId string, timeout time.Duration) (*string, error) {
+	startTime := time.Now()
+	//Wait for certificate to be issued by checking its PickupID
+	//If certID is filled then certificate should be already issued.
+
+	var certificateId string
+	for {
+		certStatus, err := c.getCertificateStatus(pickupId)
+		if err != nil {
+			return nil, fmt.Errorf("unable to retrieve: %s", err)
+		}
+		if certStatus.Status == "ISSUED" {
+			certificateId = certStatus.CertificateIdsList[0]
+			break // to fetch the cert itself
+		} else if certStatus.Status == "FAILED" {
+			return nil, fmt.Errorf("failed to retrieve certificate. Status: %v", certStatus)
+		}
+		if timeout == 0 {
+			return nil, endpoint.ErrCertificatePending{CertificateID: pickupId, Status: certStatus.Status}
+		} else {
+			log.Println("Issuance of certificate is pending...")
+		}
+		if time.Now().After(startTime.Add(timeout)) {
+			return nil, endpoint.ErrRetrieveCertificateTimeout{CertificateID: pickupId}
+		}
+		time.Sleep(2 * time.Second)
+	}
+	if certificateId == "" {
+		return nil, fmt.Errorf("something went wrong during polling cert status and we still got and empty CertificateID at the end")
+	}
+
+	return &certificateId, nil
+}
+
+func (c *Connector) getGraphqlClient() graphql.Client {
+	graphqlURL := fmt.Sprintf("%sgraphql", c.baseURL)
```

Comment:
Make the url a constant, like other endpoints

Reply:
Resolved by latest commit